sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2Suite.scala

```diff
@@ -203,33 +203,33 @@ class DataSourceV2Suite extends QueryTest with SharedSQLContext {
         val path = file.getCanonicalPath
         assert(spark.read.format(cls.getName).option("path", path).load().collect().isEmpty)
 
-        spark.range(10).select('id, -'id).write.format(cls.getName)
+        spark.range(10).select('id as 'i, -'id as 'j).write.format(cls.getName)
```

Comment:
Yes. The new resolution rule validates the dataframe that will be written to the table.

Because this uses the `DataFrameWriter` API, it matches columns by name because there isn't a strong expectation for ordering in the dataframe API (e.g. `withColumn` doesn't specify where the new column is added).